lib/_http_server.js

```diff
@@ -611,13 +611,25 @@ function onParserExecuteCommon(server, socket, parser, state, ret, d) {
   }
 }
 
+function clearIncoming(parser, req) {
+  // Reset the .incoming property so that the request object can be gc'ed.
+  if (parser && parser.incoming === req) {
+    if (req.readableEnded) {
+      parser.incoming = null;
+    } else {
+      req.on('end', clearIncoming.bind(null, parser, req));
```

Comment:
Can't we avoid the `bind()` by doing something like:

```js
function clearIncoming(req) {
  req = req || this;
  const parser = req.socket.parser;
  // Reset the .incoming property so that the request object can be gc'ed.
  if (parser && parser.incoming === req) {
    if (req.readableEnded) {
      parser.incoming = null;
    } else {
      req.on('end', clearIncoming);
    }
  }
}
```

Reply:
@mscdex Used your suggestion, PTAL